modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/parsers/PullParserFeatureReader.java

```diff
@@ -81,9 +95,26 @@ public SimpleFeature parse() throws IOException {
             throw new IOException(e);
         }
         SimpleFeature feature = (SimpleFeature) parsed;
+        if (feature != null && feature.getDefaultGeometry() != null) {
+            Geometry geometry = (Geometry) feature.getDefaultGeometry();
+            if (geometry.getUserData() instanceof CoordinateReferenceSystem) {
+                CoordinateReferenceSystem crs = (CoordinateReferenceSystem) geometry.getUserData();
+                if (WFSConfig.invertAxisNeeded(axisOrder, crs)) {
+                    try {
+                        feature.setDefaultGeometry(invertGeometryCoordinates(geometry));
+                    } catch (TransformException e) {
+                        throw new IOException(e);
+                    }
+                }
+            }
+        }
         return feature;
     }
 
+    private Object invertGeometryCoordinates(Geometry geometry) throws TransformException {
```

Comment:
Works anyways but a bit weird to have it return an Object, would look better returning a Geometry (that's what the transformer.tranfsorm method does). Not a critical issue, fix if you like.